clang/lib/Sema/SemaChecking.cpp

```diff
@@ -4185,6 +4195,52 @@ bool Sema::CheckIntelFPGABuiltinFunctionCall(unsigned BuiltinID,
   }
 }
 
+bool Sema::CheckIntelFPGAMemBuiltinFunctionCall(CallExpr *TheCall) {
+  // Make sure we have exactly 3 arguments
+  if (checkArgCount(*this, TheCall, 3))
+    return true;
+
+  Expr *PointerArg = TheCall->getArg(0);
+  QualType PointerArgType = PointerArg->getType();
+  const auto *PointerTy = dyn_cast<PointerType>(PointerArgType);
```

Comment:
Does this variable need to exist at all?  You should be able to do PointerArgType->getPointeeType() below, and isa for the 'if' statement on 4208.